lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java

```diff
@@ -245,18 +262,17 @@ public BaseEnvConfig config() {
         return config;
     }
 
-    private void prepareImage(BaseEnvConfig config, StreamQueue.LogHandle handle) {
-        handle.logOut("Pulling image {} ...", config.image());
-        final var pullingImage = retry.executeSupplier(() -> client
-            .pullImageCmd(config.image())
-            .exec(new PullImageResultCallback()));
-        try {
-            pullingImage.awaitCompletion();
-        } catch (InterruptedException e) {
-            handle.logErr("Pulling image {} was interrupted", config.image());
-            throw new RuntimeException(e);
-        }
-        handle.logOut("Pulling image {} done", config.image());
+    private void prepareImage(String image, StreamQueue.LogHandle handle) throws Exception {
+        handle.logOut("Pulling image {} ...", image);
+        AtomicInteger pullingAttempt = new AtomicInteger(0);
+        retry.executeCallable(() -> {
+            LOG.info("Pulling image {}, attempt {}", image, pullingAttempt.incrementAndGet());
+            final var pullingImage = client
+                .pullImageCmd(config.image())
+                .exec(new PullImageResultCallback());
+            return pullingImage.awaitCompletion();
+        });
+        handle.logOut("Pulling image {} done", image);
```

Comment:
Placing app log here will be suitable for measuring pulling time

Reply:
Logs with "handle.logOut" and "handle.logErr" are also printed as app logs. (Maybe they shouldn't)

Reply:
Sorry, it seems I left ambiguous definition. By 'app log' I mean usual `LOG.info(...)` and its logs which we should get in our logging UI. So I'm saying that these logs will not only be helpful for user, but for us too. For example, we'll be able to compare pulling time from dockerhub and local registry